Server/ConcordiaCurriculumManager/Controllers/CourseController.cs

```diff
@@ -129,5 +129,28 @@ public async Task<ActionResult> InitiateCourseModification([FromBody, Required]
                 detail: e.Message,
                 statusCode: StatusCodes.Status500InternalServerError);
         }
+    }
+
+    [HttpGet(nameof(GetCourseData))]
+    [SwaggerResponse(StatusCodes.Status200OK, "Course data retrieved")]
+    [SwaggerResponse(StatusCodes.Status401Unauthorized, "User is not authorized")]
+    [SwaggerResponse(StatusCodes.Status500InternalServerError, "Unexpected error")]
+    public async Task<ActionResult> GetCourseData([FromBody, Required] CourseDTO course) {

```

Comment:
It's probably easier to just make the endpoint GetCourseData/subject/catalog, and then we can just delete the CourseDTO class for simplicity.

Reply:
Sounds good